driver-core/src/main/com/mongodb/client/model/geojson/MultiPoint.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.geojson;
+
+import java.util.Collections;
+import java.util.List;
+
+import static com.mongodb.assertions.Assertions.isTrueArgument;
+import static com.mongodb.assertions.Assertions.notNull;
+
+/**
+ * A representation of a GeoJSON MultiPoint.
+ *
+ * @since 3.1
+ */
+public final class MultiPoint extends Geometry {
+
+    private final List<Position> coordinates;
```

Comment:
I understand the simplification you are making here... but this means that every position could have a different number of dimensions, which would be invalid.

Are there any run-time checks to make sure the dimensionality of all the coordinates are in agreement?